### PR TITLE
zh-cn: init the translation of HTML performance optimize

### PR DESCRIPTION
**Pull request by hanyujie2002**
### Description

Same as title
### Motivation

Ditto
### Additional details

I don't know why but there is pre commit hook which I can't pass

```
$ git commit -m "zh-ch: learn: performance: html: init translation"
[STARTED] Preparing lint-staged...
[COMPLETED] Preparing lint-staged...
[STARTED] Running tasks for staged files...
[STARTED] .lintstagedrc.json — 1 file
[STARTED] !*.md — 0 files
[STARTED] *.md — 1 file
[SKIPPED] !*.md — no files
[STARTED] markdownlint-cli2 --fix
[COMPLETED] markdownlint-cli2 --fix
[STARTED] node ./scripts/check-url-locale.js --fix
[FAILED] node ./scripts/check-url-locale.js --fix [FAILED]
[FAILED] node ./scripts/check-url-locale.js --fix [FAILED]
[COMPLETED] Running tasks for staged files...
[STARTED] Applying modifications from tasks...
[SKIPPED] Skipped because of errors from tasks.
[STARTED] Reverting to original state because of errors...
[COMPLETED] Reverting to original state because of errors...
[STARTED] Cleaning up temporary files...
[COMPLETED] Cleaning up temporary files...

✖ node ./scripts/check-url-locale.js --fix:
node:internal/errors:496
    ErrorCaptureStackTrace(err);
    ^

Error [ERR_MODULE_NOT_FOUND]: Cannot find package 'mdast-util-from-markdown' imported from C:\Users\hanyujie2002\Documents\translated-content\scripts\check-url-locale.js
    at new NodeError (node:internal/errors:405:5)
    at packageResolve (node:internal/modules/esm/resolve:887:9)
    at moduleResolve (node:internal/modules/esm/resolve:936:20)
    at defaultResolve (node:internal/modules/esm/resolve:1129:11)
    at nextResolve (node:internal/modules/esm/loader:163:28)
    at ESMLoader.resolve (node:internal/modules/esm/loader:835:30)
    at ESMLoader.getModuleJob (node:internal/modules/esm/loader:424:18)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:77:40)
    at link (node:internal/modules/esm/module_job:76:36) {
  code: 'ERR_MODULE_NOT_FOUND'
}

Node.js v18.17.1
husky - pre-commit hook exited with code 1 (error)
```

Then I bypass it by using `git commit --no-verify`

### Related issues and pull requests

Relates to #16025

### CODE REVIEW

**Comment by yin1999** [comment]
> I don't know why but there is pre commit hook which I can't pass

Have you ran `yarn install` to update the toolchain?

**Comment by hanyujie2002** [comment]
Sorry, 

> > I don't know why but there is pre commit hook which I can't pass
> 
> Have you ran `yarn install` to update the toolchain?

Sorry I can't configure this at the moment, maybe bypass the check directly is also ok ;)

File: files/zh-cn/learn/performance/html/index.md [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+
+提高性能的一个非常有用的技术是**懒加载**。这指的是在 HTML 渲染时不立即加载所有图像，而是仅在它们实际对用户可见（或即将可见）时加载它们。这意味着立即可见/可用的内容更快地准备就绪，而随后的内容只有在滚动到时才会渲染其图像，并且浏览器不会浪费带宽加载用户永远看不到的图像。
+
+懒加载通常使用 JavaScript 处理，但现在浏览器有一个可用的 `loading` 属性，可以指示浏览器自动进行图像的懒加载：
+
+```html
+<img src="800w.jpg" alt="家庭合照" loading="lazy" />
+```
+
+详细信息请参见 web.dev 上的[浏览器级 web 图像懒加载](https://web.dev/browser-level-image-lazy-loading/)。
```
Comment (yin1999) [inline files/zh-cn/learn/performance/html/index.md:141]: Could you update the url in en-US document (the previous one has been redirected)

```suggestion
详细信息请参见 web.dev 上的[浏览器级 web 图像懒加载](https://web.dev/articles/browser-level-image-lazy-loading)。
```

File: files/zh-cn/learn/performance/html/index.md [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+```html
+<video controls preload="none" poster="poster.jpg">
+  <source src="video.webm" type="video/webm" />
+  <source src="video.mp4" type="video/mp4" />
+</video>
+```
+
+将 `preload` 的值设置为 `none` 告诉浏览器在用户决定播放视频之前不要预加载任何视频数据，这对性能显然是有益的。相反，它只会显示由 `poster` 属性指示的图像。不同的浏览器具有不同的默认视频加载行为，因此最好明确指定。
+
+详细信息请参见 web.dev 上的[视频懒加载](https://web.dev/lazy-loading-video/)。
```
Comment (yin1999) [inline files/zh-cn/learn/performance/html/index.md:154]: +1

```suggestion
详细信息请参见 web.dev 上的[视频懒加载](https://web.dev/articles/lazy-loading-video)。
```

File: files/zh-cn/learn/performance/html/index.md [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+## 处理嵌入内容
+
+在网页中嵌入来自其他来源的内容非常常见。这通常在网站上显示广告以产生收入时使用，广告通常由第三方公司生成并嵌入到你的页面中。其他用途可能包括：
+
+- 显示用户可能在多个页面上需要的共享内容，例如购物车或个人资料信息。
+- 显示与组织的主要网站相关的第三方内容，例如社交媒体帖子动态。
+
+嵌入内容最常用的方式是使用 `<iframe>` 元素，尽管还存在其他不太常用的嵌入元素，如 `<object>` 和 `<embed>`。在本节中，我们将重点关注 `<iframe>`。
+
+使用 `<iframe>` 方面的最重要也是最关键的建议是：“除非你非常必要，否则不要使用嵌入式 `<iframe>`”。如果你要创建一个包含多个不同信息窗格的页面，你可能会觉得把它们分成单独的页面并加载到不同的 `<iframe>` 中是有组织性的。但是，这样做会带来一些性能和其他方面的问题：
```
Comment (yin1999) [inline files/zh-cn/learn/performance/html/index.md:165]:
```suggestion
使用 `<iframe>` 方面的最重要也是最关键的建议是：“除非非常必要，否则不要使用嵌入式 `<iframe>`”。如果你要创建一个包含多个不同信息窗格的页面，你可能会觉得把它们分成单独的页面并加载到不同的 `<iframe>` 中是有组织性的。但是，这样做会带来一些性能和其他方面的问题：
```

File: files/zh-cn/learn/performance/html/index.md [hunk 4]
```diff
@@ -0,0 +1,270 @@
...
+
+- 将内容加载到 `<iframe>` 中比将内容作为同一直接页面的一部分加载要消耗资源得多——不仅需要额外的 HTTP 请求来加载内容，而且浏览器还需要为每个 `<iframe>` 创建一个单独的页面实例。每个 `<iframe>` 实际上是嵌入在顶级网页中的一个独立网页实例。
+- 接上一点，你还需要为每个不同的 `<iframe>` 分别处理任何 CSS 样式或 JavaScript 操作（除非嵌入的页面来自同一源），这变得更加复杂。你无法使用应用于顶级页面的 CSS 和 JavaScript 来定位嵌入的内容，反之亦然。这是一项合理的安全措施，是 Web 的基本原则。想象一下，如果第三方嵌入的内容可以任意在其嵌入的任何页面上运行脚本，那么你可能会遇到的问题！
+- 每个 `<iframe>` 还需要单独加载所有共享的数据和媒体文件——你无法在不同的页面嵌入之间共享缓存的资源（再次强调，除非嵌入的页面来自同一源）。这可能导致页面使用的带宽比你预期的要多得多。
+
+建议将内容放在单个页面中。如果你想在页面更改时动态获取新内容，仍然比将其放入 `<iframe>` 中更好，这样可以提高性能。例如，你可以使用 {{domxref("fetch()")}} 方法获取新数据，然后使用一些 DOM 脚本将其插入到页面中。更多信息，请参见[从服务器获取数据](/zh-CN/docs/Learn/JavaScript/Client-side_web_APIs/Fetching_data)和[操作文档](/zh-CN/docs/Learn/JavaScript/Client-side_web_APIs/Manipulating_documents)。
+
+> **备注：** 如果你掌控内容并且内容相对简单，你可以考虑使用在 `src` 属性中以 base-64 编码的内容填充 `<iframe>`，甚至可以将原始 HTML 插入到 `srcdoc` 属性中（有关更多信息，请参见 [Iframe 性能第二部分：好消息](https://medium.com/slices-of-bread/iframe-performance-part-2-the-good-news-26eb53cea429)）。
+
+如果必须使用 `<iframe>`，请谨慎使用。
```
Comment (yin1999) [inline files/zh-cn/learn/performance/html/index.md:175]:
```suggestion
如果必须使用 `<iframe>`，请保持谨慎。
```

File: files/zh-cn/learn/performance/html/index.md [hunk 5]
```diff
@@ -0,0 +1,270 @@
...
+#### 懒加载 iframe
+
+与 `<img>` 元素一样，你还可以使用 `loading` 属性指示浏览器对最初屏幕外的 `<iframe>` 内容进行懒加载，从而提高性能：
+
+```html
+<iframe src="https://example.com" loading="lazy" width="600" height="400">
+</iframe>
+```
+
+详情请参阅[是时候懒加载屏外 iframe 了！](https://web.dev/iframe-lazy-loading/)。
```
Comment (yin1999) [inline files/zh-cn/learn/performance/html/index.md:186]: +1

```suggestion
详情请参阅[是时候延迟加载屏外 iframe 了！](https://web.dev/articles/iframe-lazy-loading)。
```

File: files/zh-cn/learn/performance/html/index.md [hunk 6]
```diff
@@ -0,0 +1,270 @@
...
+
+资源加载的顺序对于最大化感知和实际性能非常重要。当加载网页时：
+
+1. 通常首先解析 HTML，按照页面上的顺序进行解析。
+2. 解析任何找到的 CSS，以理解应该应用于页面的样式。在此期间，开始获取链接的资源，如图像和 Web 字体。
+3. 解析任何找到的 JavaScript，对页面进行评估和执行。默认情况下，这会阻止解析遇到的 {{htmlelement("script")}} 元素之后的 HTML。
+4. 稍后，浏览器根据应用于它的 CSS 来确定每个 HTML 元素应该如何进行样式化。
+5. 将经过样式化的结果绘制到屏幕上。
+
+> **注意：** 这只是一个非常简单的叙述，但可以让你了解发生的事情。
```
Comment (yin1999) [inline files/zh-cn/learn/performance/html/index.md:198]:
```suggestion
> **备注：** 这只是一个非常简单的叙述，但可以让你了解发生的事情。
```

File: files/zh-cn/learn/performance/html/index.md [hunk 7]
```diff
@@ -0,0 +1,270 @@
...
+});
+```
+
+我们可以通过在 `<script>` 元素中引用它来将此脚本应用于页面：
+
+```html
+<script src="index.js"></script>
+```
+
+如果我们将这个 `<script>` 元素放在 `<p>` 元素之前（例如，在 {{htmlelement("head")}} 元素中），页面会抛出错误（例如，Chrome 报告到控制台上的错误为 "Uncaught TypeError: Cannot read properties of null (reading 'addEventListener')"）。这是因为脚本依赖于 `<p>` 元素才能正常工作，但在解析脚本时，`<p>` 元素在页面上并不存在。它尚未被呈现。
```
Comment (yin1999) [inline files/zh-cn/learn/performance/html/index.md:230]:
```suggestion
如果我们将这个 `<script>` 元素放在 `<p>` 元素之前（例如，在 {{htmlelement("head")}} 元素中），页面会抛出错误（例如，Chrome 报告到控制台上的错误为“Uncaught TypeError: Cannot read properties of null (reading 'addEventListener')”）。这是因为脚本依赖于 `<p>` 元素才能正常工作，但在解析脚本时，`<p>` 元素在页面上并不存在。它尚未被渲染。
```

File: files/zh-cn/learn/performance/html/index.md [hunk 8]
```diff
@@ -0,0 +1,270 @@
...
+
+要解决上述问题，可以将 `<script>` 元素放在 `<p>` 元素之后（例如放在文档主体的末尾），或者在适当的事件处理程序中运行代码（例如，在 [`DOMContentLoaded`](/zh-CN/docs/Web/API/Document/DOMContentLoaded_event) 上运行，其在 DOM 完全解析后触发）。
+
+然而，这并不能解决等待脚本加载的问题。可以通过在 `<script>` 元素中添加 `async` 属性来实现更好的性能：
+
+```html
+<script async src="index.js"></script>
+```
+
+这会使脚本获取与 DOM 解析并行进行，因此它会在相同的时间准备好，不会阻塞呈现，从而提高性能。
```
Comment (yin1999) [inline files/zh-cn/learn/performance/html/index.md:240]:
```suggestion
这会使脚本获取与 DOM 解析并行进行，因此它会在相同的时间准备好，不会阻塞渲染，从而提高性能。
```

File: files/zh-cn/learn/performance/html/index.md [hunk 9]
```diff
@@ -0,0 +1,270 @@
...
+
+> **注意：** 还有另一个属性 `defer`，它会使脚本在文档解析完成之后、`DOMContentLoaded` 事件触发之前执行。这与 `async` 有类似的效果。
```
Comment (yin1999) [inline files/zh-cn/learn/performance/html/index.md:242]:
```suggestion
> **备注：** 还有另一个属性 `defer`，它会使脚本在文档解析完成之后、`DOMContentLoaded` 事件触发之前执行。这与 `async` 有类似的效果。
```

File: files/zh-cn/learn/performance/html/index.md [hunk 10]
```diff
@@ -0,0 +1,270 @@
...
+
+处理 JavaScript 加载的另一个技巧是将脚本拆分为代码模块，并在需要时加载每个部分，而不是将所有代码放入一个巨大的脚本并在开头加载。这可以通过使用 [JavaScript 模块](/zh-CN/docs/Web/JavaScript/Guide/Modules) 来实现。阅读链接的文章以获取详细指南。
```
Comment (yin1999) [inline files/zh-cn/learn/performance/html/index.md:244]:
```suggestion
处理 JavaScript 加载的另一个技巧是将脚本拆分为代码模块，并在需要时加载每个部分，而不是将所有代码放入一个巨大的脚本并在开头加载。这可以通过使用 [JavaScript 模块](/zh-CN/docs/Web/JavaScript/Guide/Modules)来实现。阅读链接的文章以获取详细指南。
```

File: files/zh-cn/learn/performance/html/index.md [hunk 11]
```diff
@@ -0,0 +1,270 @@
...
+```html
+<link rel="preload" href="sintel-short.mp4" as="video" type="video/mp4" />
+```
+
+遇到 `rel="preload"` 链接时，浏览器将尽快获取所引用的资源，并使其在浏览器缓存中可用，以便在后续代码中引用时更快地准备好。预加载用户在页面浏览早期会遇到的高优先级资源非常有用，这样可以让用户获得尽可能流畅的体验。
+
+有关使用 `rel="preload"` 的详细信息，请参阅以下文章：
+
+- [`rel="preload"`](/zh-CN/docs/Web/HTML/Attributes/rel/preload)
+- [预加载关键资源以提高加载速度](https://web.dev/preload-critical-assets/) web.dev（2020）
```
Comment (yin1999) [inline files/zh-cn/learn/performance/html/index.md:259]:
```suggestion
- [预加载关键资源以提高加载速度](https://web.dev/articles/preload-critical-assets) web.dev（2020）
```

**Review by yin1999 — CHANGES_REQUESTED** [review]

**Review by yin1999 — APPROVED** [review]
Thanks

**Pull request merged**